a2s/a2s_protocol.py

```diff
@@ -0,0 +1,22 @@
+__all__ = ("A2SProtocol",)
+
+
+from typing import Any, Optional
+
+from .byteio import ByteReader
+
+
+class A2SProtocol:
+    @staticmethod
+    def serialize_request(challenge: int) -> bytes:
+        raise NotImplemented
```

Comment:
That's raising the [NotImplemented constant](https://docs.python.org/3/library/constants.html#NotImplemented), not the NotImplementedError. It also needs to be instantiated before raising.